src/test/java/org/prebid/server/settings/CompositeApplicationSettingsTest.java

```diff
@@ -0,0 +1,303 @@
+        assertThat(future.succeeded()).isTrue();
+        assertThat(future.result()).isSameAs(account);
+    }
+
+    @Test
+    public void getAccountByIdShouldReturnEmptyResultIfAllDelegatesFail() {
+        // given
+        given(delegate1.getAccountById(anyString(), any()))
+                .willReturn(Future.failedFuture(new PreBidException("error1")));
+
+        given(delegate2.getAccountById(anyString(), any()))
+                .willReturn(Future.failedFuture(new PreBidException("error2")));
+
+        // when
+        final Future<Account> future = compositeApplicationSettings.getAccountById("ignore", null);
+
+        // then
+        assertThat(future.failed()).isTrue();
+        assertThat(future.cause().getMessage()).isEqualTo("error2");
+    }
+
+    @Test
+    public void getAdUnitConfigByIdShouldReturnConfigFromFirstDelegateIfPresent() {
+        // given
+        given(delegate1.getAdUnitConfigById(anyString(), any()))
+                .willReturn(Future.succeededFuture("adUnitConfig1"));
+
+        // when
+        final Future<String> future = compositeApplicationSettings.getAdUnitConfigById("ignore", null);
+
+        // then
+        assertThat(future.succeeded()).isTrue();
+        assertThat(future.result()).isEqualTo("adUnitConfig1");
+        verifyZeroInteractions(delegate2);
+    }
+
+    @Test
+    public void getAdUnitConfigByIdShouldReturnConfigFromSecondDelegateIfFirstDelegateFails() {
+        // given
+        given(delegate1.getAdUnitConfigById(anyString(), any()))
+                .willReturn(Future.failedFuture(new PreBidException("error1")));
+
+        given(delegate2.getAdUnitConfigById(anyString(), any()))
+                .willReturn(Future.succeededFuture("adUnitConfig1"));
+
+        // when
+        final Future<String> future = compositeApplicationSettings.getAdUnitConfigById("ignore", null);
+
+        // then
+        assertThat(future.succeeded()).isTrue();
+        assertThat(future.result()).isEqualTo("adUnitConfig1");
+    }
+
+    @Test
+    public void getAdUnitConfigByIdShouldReturnEmptyResultIfAllDelegatesFail() {
+        // given
+        given(delegate1.getAdUnitConfigById(anyString(), any()))
+                .willReturn(Future.failedFuture(new PreBidException("error1")));
+
+        given(delegate2.getAdUnitConfigById(anyString(), any()))
+                .willReturn(Future.failedFuture(new PreBidException("error2")));
+
+        // when
+        final Future<String> future = compositeApplicationSettings.getAdUnitConfigById("ignore", null);
+
+        // then
+        assertThat(future.failed()).isTrue();
+        assertThat(future.cause().getMessage()).isEqualTo("error2");
+    }
+
+    @Test
+    public void getStoredRequestsByIdReturnResultFromFirstDelegateIfPresent() {
+        // given
+        given(delegate1.getStoredRequestsById(anySet(), any()))
+                .willReturn(Future.succeededFuture(StoredRequestResult.of(singletonMap("k1", "v1"), emptyList())));
```

Comment:
It's a matter of taste of course but "key1" and "value1" require less thinking to understand what they stand for than "k1" and "v1"